Sources/MlemMiddleware/API Client/ApiClient+Post.swift

```diff
@@ -266,7 +266,8 @@ public extension ApiClient {
         let request = EditPostRequest(
             postId: id,
             name: title,
-            url: linkUrl?.absoluteString,
+            // Submitting `nil` doesn't remove the link, but submitting an empty string does.
+            url: linkUrl?.absoluteString ?? "",
```

Comment:
This causes a 400 error against sh.itjust.works (and local 19.6):
<img width="1004" alt="Screenshot 2024-09-20 at 9 55 46 AM" src="https://github.com/user-attachments/assets/f4918c7b-a705-4d25-9379-0d66ff6dfaf8">

Submitting with `null` does work as intended from Postman.